stan/math/gpu/kernels/check_symmetric_kernel.cl

```diff
@@ -0,0 +1,19 @@
+R"(
+#ifndef A
+#define A(i,j)  A[j*rows+i]
+#endif
+__kernel void check_symmetric(
+      __global double *A,
+      int rows,
+      int cols,
+      __global int *flag,
+      double tolerance) {
+  const int i = get_global_id(0);
+  const int j = get_global_id(1);
+  if( i < rows && j < cols ) { 
+    double diff = fabs(A(i,j)-A(j,i));
+    if ( diff > tolerance ) {
+      flag[0] = 1;
```

Comment:
This is going to lead to a race condition for flag[0] right?

I think it's fine for now, but we can look at this again when we want to improve performance

Reply:
There is no read of the flag[0] value and as long as any thread sets it (no thread will clear it at any time), this isnt really a hazard. But I will rewrite this with an atomic instruction. It wont hurt performance, so its not really a problem.

Reply:
Yeah it's only possibly bad if you had a big matrix that fails lots of `(diff > tolerance)` as the writes pile up, but idt it would even be that bad for the reasons you stated above. If you don't think this will incure a noticable cost in that instance then let's leave this as is.

Reply:
I am sure it will not. It would be an issue off course if we wanted a count of NaN values, but that is not the case here.